### ISSUE TITLE
Fatal error: Class 'Slim\Middleware\JwtAuthentication' not found 

### ISSUE THREAD

**Issue by fcolasante** [tuupola/slim-jwt-auth#97]
**Middleware.php**
```php

// Application middleware

// e.g: $app->add(new \Slim\Csrf\Guard);
use Tuupola\Middleware\HttpBasicAuthentication;

$container = $app->getContainer();
$container['logger'] = function($c) {
 $logger = new \Monolog\Logger('my_logger');
 $file_handler = new \Monolog\Handler\StreamHandler("../logs/app.log");
 $logger->pushHandler($file_handler);
 return $logger;
};

$container["jwt"] = function ($container) {
    return new StdClass;
   };

   $app->add(new \Slim\Middleware\JwtAuthentication([
    "path" => "/",
    "logger" => $container[‘logger’],
    "secret" => "123456789helo_secret",
    "rules" => [
    new \Slim\Middleware\JwtAuthentication\RequestPathRule([
    "path" => "/",
    "passthrough" => ["/token", "/not-secure", "/home"]
    ]),
    new \Slim\Middleware\JwtAuthentication\RequestMethodRule([
    "passthrough" => ["OPTIONS"]
    ]),
    ],
    "callback" => function ($request, $response, $arguments) use ($container) {
    $container["jwt"] = $arguments["decoded"];
    },
    "error" => function ($request, $response, $arguments) {
    $data["status"] = "error";
    $data["message"] = $arguments["message"];
    return $response
    ->withHeader("Content-Type", "application/json")
    ->write(json_encode($data, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT));
    }
   ]));
```

Where is the problem? I Slim 3 and composer

**Comment by tuupola** [comment]
How did you install the middleware? Which version the middleware you have installed?

**Comment by fcolasante** [comment]
I' ve installed the dependencies  with composer and then i've copied this code at middleware.php, i might install something, i don't found any good tutorial

**Comment by tuupola** [comment]
You can see which version you installed from `composer.json` file.

**Comment by fcolasante** [comment]
This is **Composer.json**:
```
{
    "name": "slim/slim-skeleton",
    "description": "A Slim Framework skeleton application for rapid development",
    "keywords": ["microframework", "rest", "router", "psr7"],
    "homepage": "http://github.com/slimphp/Slim-Skeleton",
    "license": "MIT",
    "authors": [
        {
            "name": "Josh Lockhart",
            "email": "info@joshlockhart.com",
            "homepage": "http://www.joshlockhart.com/"
        }
    ],
    "require": {
        "php": ">=5.5.0",
        "slim/slim": "^3.1",
        "slim/php-view": "^2.0",
        "monolog/monolog": "^1.17",
        "tuupola/slim-jwt-auth": "^2.3",
        "tuupola/cors-middleware": "^0.5.2",
        "firebase/php-jwt": "^5.0",
        "tuupola/base62": "^0.9.0",
        "tuupola/slim-basic-auth": "^2.3"
    },
    "require-dev": {
        "phpunit/phpunit": ">=4.8 < 6.0"
    },
    "autoload": {
        "psr-4": {
            "tuupola/slim-jwt-auth\\": "src/"}
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "config": {
        "process-timeout" : 0
    },
    "scripts": {
        "start": "php -S localhost:8080 -t public index.php",
        "test": "phpunit"
    }

}
```

**Comment by tuupola** [comment]
Remove the following code which has been added to `composer.json` manually. Composer itself has not added it.

```
"autoload": {
        "psr-4": {
            "tuupola/slim-jwt-auth\\": "src/"
        }
},
```

Have you actually installed the dependencies or just added them to `composer.json` file?

**Comment by fcolasante** [comment]
I used composer require and composer install for installing all dependencies
I've tried to insert manually this loading. Before I wrote this line of code, anyway the code still not wasn't working

**Comment by tuupola** [comment]
There is no need to edit `composer.json` file. To install `slim/slim-skeleton` and `tuupola/slim-jwt-auth` run the following commands.

```
$ composer create-project slim/slim-skeleton something
$ cd something
$ composer require tuupola/slim-jwt-auth
```

**Comment by fcolasante** [comment]
Does it depend on php 7?

**Comment by tuupola** [comment]
You can see the requirements in the package info page:

https://packagist.org/packages/tuupola/slim-jwt-auth#3.0.0-rc.5
https://packagist.org/packages/tuupola/slim-jwt-auth#2.3.3

TL;DR the unreleased 3.x branch requires PHP 7.1 while the current stable 2.x branch requires PHP 5.5.

**Comment by fcolasante** [comment]
I tried in every possible mode but there are always the same error. 
I have php 5.5, slim-skeleton and tuupola/slim-jwt-auth.
When I put in middleware.php this code:
`   $app->add(new \Slim\Middleware\JwtAuthentication([
    "path" => "/",
    "logger" => $container[‘logger’],
    "secret" => "123456789helo_secret",
    "rules" => [
    new \Slim\Middleware\JwtAuthentication\RequestPathRule([
    "path" => "/",
    "passthrough" => ["/token", "/not-secure", "/home"]
    ]),
    new \Slim\Middleware\JwtAuthentication\RequestMethodRule([
    "passthrough" => ["OPTIONS"]
    ]),
    ],
    "callback" => function ($request, $response, $arguments) use ($container) {
    $container["jwt"] = $arguments["decoded"];
    },
    "error" => function ($request, $response, $arguments) {
    $data["status"] = "error";
    $data["message"] = $arguments["message"];
    return $response
    ->withHeader("Content-Type", "application/json")
    ->write(json_encode($data, JSON_UNESCAPED_SLASHES | JSON_PRETTY_PRINT));
    }
   ]));`

I have this error: 
`Fatal error: Class 'Slim\Middleware\HttpBasicAuthentication' not found in`

**Comment by tuupola** [comment]
Sorry I have no idea what is wrong. 

Look at `composer.json` to see what and which version you have installed. Look at `vendor/` folder to confirm which libraries and which versions have really been installed. In case of problems make sure you have latest composer. Look at the line number of the error message and then look at that specific file and line. In other words all the usual debugging stuff. 

Still does not help? Start from scratch, don't just copypaste code without understanding what it does.  Install one composer package at time to see where the problem appears.

[Composer documentation](https://getcomposer.org/) for is good for reading about composer usage and [StackOverflow](https://stackoverflow.com/) for general help.

The error message is not about `JwtAuthentication` it is about `HttpBasicAuthentication`. Have you installed the basic auth middleware?

**Comment by BuilderQ** [comment]
I ran into same issue and my solution ended up being that I had to change namespace from `new \Slim\Middleware\HttpBasicAuthentication` to `new \Tuupola\Middleware\HttpBasicAuthentication`. It seems that some examples in Internet had wrong class name.

**Comment by tuupola** [comment]
@BuilderQ see [UPGRADING](https://github.com/tuupola/slim-jwt-auth/blob/3.x/UPGRADING.md) for more info.
